src/moonlink_service/src/test.rs

```diff
@@ -178,10 +214,38 @@ async fn read_all_batches(url: &str) -> Vec<RecordBatch> {
     reader.into_iter().map(|b| b.unwrap()).collect()
 }
 
+#[tokio::test]
+#[serial]
+async fn test_health_check_endpoint() {
+    let _guard = TestGuard::new(&get_moonlink_backend_dir()).await;
+    let config = get_service_config();
+    tokio::spawn(async move {
+        start_with_config(config).await.unwrap();
+    });
+    test_readiness_probe().await;
+
+    let client = reqwest::Client::new();
+    let response = client
+        .get(format!("{REST_ADDR}/health"))
+        .header("content-type", "application/json")
+        .send()
+        .await
+        .unwrap();
+
+    assert!(
+        response.status().is_success(),
+        "Response status is {response:?}"
+    );
+    let health_data: serde_json::Value = response.json().await.unwrap();
```

Comment:
Same question here, I would prefer if we could get a concrete type and access its data member.